docs/project_practice/how_to_build_a_todo_application_with_ribir.md

```diff
@@ -4,17 +4,17 @@ sidebar_position: 1
 
 # How to build a Todo Application with Ribir
 
-In this tutorial, We'll start learning how to build a Todo application using Ribir. By following this tutorial in its entirety, you will be able to complete a Todo application like this by yourself:
+In this tutorial, we will learn how to build a Todo application using Ribir. By following this tutorial in its entirety, you will be able to complete a Todo application like this by yourself:
 
 ![todo example](./img/todo_example.png)
 
 This will be an exciting journey. Let's go!
 
 ## Define the Application data structure
 
-The data structure is the core of the application. In Ribir the interface is just the presentation around the data structure, not interfering with the data structure. So we need to define Todo data structure first.
+The data structure is the core of the application. In Ribir the interface is just the presentation around the data structure, but not interfering with the data structure. So we need to define the Todo data structure first.
 
-Now let's define the simplest Todo application data structure. I call this structure `TodoMVP`. And as a simplest Todo application has a list of what it use to save the collection of todo task at least. In Rust, we can use `Vec` structure to describe a list. And then, the list must have many task. A single task needs two fields, `done` and `description`. The `done` field use type `bool` to describe task whether or not to complete, and the `description` field use type `String` to describe task specific information.
+Now let's define the simplest Todo application data structure. I call this structure `TodoMVP`. As a simplest Todo application it has a list of what it uses to save the collection of todo tasks at least. In Rust, we can use the `Vec` structure to describe a list, and then, the list must be able hold many task. A single task needs two fields, `done` and `description`. The `done` field is of type `bool` to describe whether or not the task is completed and the `description` field is of type `String` to describe task specific information.
```

Comment:
the list must be able hold many task --> the list must be able to hold many task